test/compilable/test7616.d

```diff
@@ -0,0 +1,11 @@
+module test7616;
+
+@safe pure nothrow @nogc:
+void foo() {}
+struct Test
+{
+    static void bar() {}
+}
+
+static assert(is(typeof(&foo) == typeof(&Test.bar)));
```

Comment:
Please use the test in the original bug report. This test doesn't even test the corect propagation but rather that foo and bar have the same signature (including attributes). That means that it doesn't test anything regarding the initial issue. If you delete the `@safe pure nothrow nogc:` line, the test is still going to pass.

Reply:
I will not use the test from the original bug report because that relies on manually examining the output of pragma(msg) and doesn't cover @nogc. I did, however, change it to express the expected type explicitly, and to check both static and non-static functions.